pkg/csi/cinder/nodeserver.go

```diff
@@ -411,6 +411,32 @@ func (ns *nodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		}
 	}
 
+	// Try expanding the volume if it's created from a snapshot or another volume (see #1539)
+	if vol.SourceVolID != "" || vol.SnapshotID != "" {
+
+		r := mountutil.NewResizeFs(ns.Mount.Mounter().Exec)
+
+		needResize, err := r.NeedResize(devicePath, stagingTarget)
+
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not determine if volume %q need to be resized: %v", volumeID, err)
+		}
+
+		if needResize {
+			sourceID := vol.SourceVolID
+			sourceType := "source volume"
+			if vol.SourceVolID == "" {
```

Comment:
Is this supposed to be `vol. SnapshotID`?

Reply:
I guess we can remove the lines from L426-L432 . We can just add a log statement with volumeID which should suffice.

Reply:
@lingxiankong oh, my mistake, thanks.
@ramineni Sure, I'll remove them.

Reply:
done